scanner.go

```diff
@@ -62,9 +65,10 @@ var ScanAllKeys = ScanOpts{
 type scanner struct {
 	Client
 	ScanOpts
-	res []string
-	cur string
-	err error
+	res    scanResult
+	resIdx int
+	cur    string
```

Comment:
is `cur` still needed here? it looks like it got moved into `scanRes`

Reply:
Done